### ISSUE TITLE
Migration using typescript

### ISSUE THREAD

**Issue by thevirajshelke** [tgriesser/knex#3194]
# Environment

Knex version: 0.16.5
Database + version: postgreSQL 10.8
OS: Linux
Node Version: v10.15.3

**knexfile.ts**
```ts
import * as path from "path";
import * as knex from "knex";

const database = {
  client: "sqlite3",
  connection: {
    filename: path.join(__dirname, "dev.sqlite3")
  },
  useNullAsDefault: true,
  migrations: {
    directory: path.join(__dirname, "./migrations")
  }
} as knex.Config;

export = database;
```
**connection.ts**
```ts
import * as Knex from "knex";
import * as database from "./knexfile";
export const knex = Knex(database);
```

**migration.ts**
```ts
/* eslint-disable no-unused-vars */

export const up = (knex: any, promise: Promise<any>) => {
  return knex.schema.createTable("test", (table: any) => {
    // some code
  });
};

export const down = (knex: any, promise: Promise<any>) => {
  return knex.schema.dropTable("test");
};

```

**main.ts**
```ts
import { knex } from "./database/connection";

(async () => {
  await knex.migrate.latest();
})();
```

The above code is giving me error as follows
```bash
(function (exports, require, module, __filename, __dirname) { export declare const up: (knex: any, promise: Promise<any>) => any;
                                                              ^^^^^^

SyntaxError: Unexpected token export
```

Any suggestions on how should you write migration scripts in typescript?
I am compiling my source to javascript and then using it.

**Comment by kibertoad** [comment]
@thevirajshelke Are you compiling your migrations as well? Because error is clearly stating that you are trying to run non-transpiled code. You can use ts-node for that.

**Comment by thevirajshelke** [comment]
> @thevirajshelke Are you compiling your migrations as well? Because the error is clearly stating that you are trying to run the non-transpiled code. You can use ts-node for that.

@kibertoad Yes, I am compiling the migration files as well and generating `js` file. Like you pointed out the error states that I am trying to run the non-transpiled code, so I just checked and found out that my `tsc` is generating `*.d.ts` files as well! And somehow how knex while migration reads these declaration files as well! **So when I deleted the `*.d.ts` file and ran the same code migrations worked fine!**

So what do you think? Is it a bug? I mean knex must be trying to read all the files in the migration folder & in this case it shouldn't. So maybe can we tell knex to ignore `*.d.ts` files somehow, maybe in the config?

Migration folder also contains `*.js.map` file which knex doesn't read! So just the way it ignores `*.js.map` it should ignore `*.d.ts` as well!

**Comment by kibertoad** [comment]
There is extensions config param for this purpose, I believe.

**Comment by thevirajshelke** [comment]
**I just referred docs**

**_loadExtensions:_** array of file extensions which knex will treat as migrations. For example, if you have typescript transpiled into javascript in the same folder, you want to execute only javascript migrations. In this case, set loadExtensions to ['.js'] (Notice the dot!) (default ['.co', '.coffee', '.eg', '.iced', '.js', '.litcoffee', '.ls', '.ts'])

**_extension:_** the file extension used for the generated migration files (default js)

So I changed my main.ts file as follows
```ts
import { knex } from "./database/connection";

(async () => {
  try {
    console.log("Migration started...");
    await knex.migrate.latest({ loadExtensions: [".js"] });
    console.log("Migration finished...");
  } catch (err) {
    console.log(err);
  }
})();
```

but my `tsc` gives me this error
```bash
"Argument of type '{ loadExtensions: string[]; }' is not assignable to parameter of type 'MigratorConfig'.\n  Object literal may only specify known properties, and 'loadExtensions' does not exist in type 
```

**But when I write `@ts-ignore` just to check if the code works fine, it works fine!**
```ts
import { knex } from "./database/connection";

(async () => {
  try {
    console.log("Migration started...");
    // @ts-ignore
    await knex.migrate.latest({ loadExtensions: [".js"] });
    console.log("Migration finished...");
  } catch (err) {
    console.log(err);
  }
})();
```
So this means I can tell knex to ignore `*.d.ts` file by using `{ loadExtensions: [".js"] }` config param but It seems that typescript typings/declarations are missing for `MigratorConfig`

**Comment by kibertoad** [comment]
It was fixed in master, you can use 0.17.0-next

**Comment by thevirajshelke** [comment]
Okay sounds cool! :) I'll close this issue then :+1: 
@kibertoad Thanks for all the help :+1:
